src/main/java/com/moneylendingapp/services/impl/DefaultUserServiceImpl.java

```diff
@@ -61,4 +76,45 @@ private void ensureUsernameIsUnique(String username) {
         }
     }
 
+    @Override
+    public LoginResponse login(LoginRequest loginRequest) throws Exception {
+
+        User user = userRepo.findByUsername(loginRequest.getUsername())
+                .orElseThrow(() ->
+                        new UsernameNotFoundException(String.format("Username: %s not found", loginRequest.getUsername())));
+
+        return authenticateUser(loginRequest, user);
+    }
+
+    private LoginResponse authenticateUser(LoginRequest loginRequest, User user) throws Exception {
+        try {
+            authenticationManager.authenticate(
+                    new UsernamePasswordAuthenticationToken(
+                            loginRequest.getUsername(), loginRequest.getPassword()));
+
+            UserDetails userDetails = userDetailsService.loadUserByUsername(loginRequest.getUsername());
+            String jwtToken = jwtTokenUtil.generateToken(userDetails);
+            log.info("Token = {}", jwtToken);
+
+            return LoginResponse.builder()
+                    .id(user.getId())
+                    .token(jwtToken)
+                    .role(user.getRoles())
+                    .build();
+
+        } catch (Exception ex) {
+            log.error("Authentication failed for {}. {}", loginRequest.getUsername(), ex.getMessage());
+            throw new Exception(ex.getMessage());
+        }
+    }
+
+    public User getLoggedInUser(){
+
+        return userRepo.findByUsername(
+                SecurityContextHolder.getContext()
+                        .getAuthentication()
+                        .getName()).orElseThrow(()
```

Comment:
Move this to a new line `  SecurityContextHolder.getContext()
                        .getAuthentication()
                        .getName()` and pass the variable into findByUsername()